eitprocessing/parameters/eeli.py

```diff
@@ -33,20 +41,63 @@ def compute_parameter(self, continuous_data: ContinuousData) -> np.ndarray:
+            sequence: optional, Sequence to store the result in.
+            store: whether to store the result in the sequence, defaults to `True` if a Sequence if provided.
+            result_label: label of the returned SparseData object, defaults to `'continuous_eelis'`.
 
         Returns:
-            np.ndarray: the end-expiratory values of all breaths in the impedance data.
+            A SparseData object with the end-expiratory values of all breaths in the impedance data.
+
+        Raises:
+            RuntimeError: If store is set to true but no sequence is provided.
+            ValueError: If the provided sequence is not an instance of the Sequence dataclass.
+            ValueError: If tiv_method is not one of 'inspiratory', 'expiratory', or 'mean'.
         """
+        if store is None and isinstance(sequence, Sequence):
+            store = True
+
+        if store and sequence is None:
+            msg = "Can't store the result if no Sequence is provided."
+            raise RuntimeError(msg)
+
+        if store and not isinstance(sequence, Sequence):
+            msg = "To store the result a Sequence dataclass must be provided."
+            raise ValueError(msg)
+
         check_category(continuous_data, "impedance", raise_=True)
 
         bd_kwargs = self.breath_detection_kwargs.copy()
         breath_detection = BreathDetection(**bd_kwargs)
         breaths = breath_detection.find_breaths(continuous_data)
 
         if not len(breaths):
-            return np.array([], dtype=float)
+            return SparseData(
+                label=result_label,
+                name="End-expiratory lung impedance (EELI)",
+                unit=None,
+                category="impedance",
+                time=np.array([], dtype=float),
+                description="End-expiratory lung impedance (EELI) determined on continuous data",
+                parameters=self.breath_detection_kwargs,
+                derived_from=[continuous_data],
+                values=np.array([], dtype=float),
+            )
 
         _, _, end_expiratory_times = zip(*breaths.values, strict=True)
         end_expiratory_indices = np.flatnonzero(np.isin(continuous_data.time, end_expiratory_times))
 
-        return continuous_data.values[end_expiratory_indices]
+        eeli_container = SparseData(
+            label=result_label,
+            name="End-expiratory lung impedance (EELI)",
+            unit=None,
+            category="impedance",
+            time=[breath.end_time for breath in breaths.values if breath is not None],
+            description="End-expiratory lung impedance (EELI) determined on continuous data",
+            parameters=self.breath_detection_kwargs,
+            derived_from=[continuous_data],
+            values=continuous_data.values[end_expiratory_indices],
+        )
+        if store:
+            sequence.sparse_data.add(eeli_container)
+
+        return eeli_container
```

Comment:
These is a lot of repeated code here. Could you instead put the creation of the time and values within a `if`/`else` block, and create the `SparseData` object once?